zerver/lib/markdown/__init__.py

```diff
@@ -125,27 +125,22 @@ def normal_compile(pattern: str) -> Any:
     )
 
 
-STREAM_LINK_REGEX = r"""
-                     (?<![^\s'"\(,:<])            # Start after whitespace or specified chars
-                     \#\*\*                       # and after hash sign followed by double asterisks
-                         (?P<stream_name>[^\*]+)  # stream name can contain anything
-                     \*\*                         # ends by double asterisks
-                    """
-
-
-@one_time
-def get_compiled_stream_link_regex() -> Pattern:
-    return verbose_compile(STREAM_LINK_REGEX)
```

Comment:
@showell do you remember the context around adding this function in ad071ced4782aba135b59db8597fa4f75f0e91a2?  I think it was part of a performance optimization effort, and we should verify that we're not regressing performance by removing that pre-compiling.

Reply:
I have updated this so that we are still pre-compiling the regex to avoid concerned regressions related to this. But `verbose_compile` is no longer used for pre-compiling `STREAM_LINK_REGEX`, `STREAM_TOPIC_LINK_REGEX` as it adds ^(.*?) and (.*?)$ which cause extra overhead of matching pattern which is not required with new `InlineProcessor `[(docs mentioning this)](https://python-markdown.github.io/extensions/api/#comparison-with-new-inlineprocessor) these extra patterns at beggining and end are not required. So, `StreamPattern` and `StreamTopicPattern` now define their own `__init__` method for accepting pre-compiled the regex.
As we cannot use `CompiledPattern` class for this(because it inherits `markdown.inlinepatterns.Pattern` which is older API and reason for this bug), so I had to define their separate `__init__` constructor.